### PR TITLE
Add support for LXD virtual-machine type images

### PR DESCRIPTION
**Pull request by hasusuf**
Virtual machines are natively supported by LXD starting from version [4.0](https://discuss.linuxcontainers.org/t/running-virtual-machines-with-lxd-4-0/7519)

This PR adds an option to create image type `VIRTUAL-MACHINE` if `true`; defaults to `false` for `CONTAINER`.

### CODE REVIEW

File: builder/lxd/step_lxd_launch.go
Comment (sylviamoss) [inline builder/lxd/step_lxd_launch.go:24]: This is not needed. 
```suggestion
```

**Comment by sylviamoss** [comment]
Hello, @ChrisLundquist! As the codeowner, do you mind reviewing this one?

**Review by ChrisLundquist — APPROVED** [review]

**Comment by hasusuf** [comment]
@sylviamoss @ChrisLundquist One of you guys should merge this PR I don't have `write access`, thanks!

**Review by nywilken — APPROVED** [review]
@hasusuf this looks good thanks for the contribution. I'm going to add a follow up PR immediately after the merge to include the auto-generated builder config partials into 'docs/builders/lxd.mdx'

**Comment by nywilken** [comment]
@ChrisLundquist thanks for the assist on the review. Cheers!

**Pull request merged**
